nemo/collections/asr/data/audio_to_audio.py

```diff
@@ -0,0 +1,891 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+import math
+import random
+from collections import OrderedDict
+from typing import Dict, Iterable, List, Optional, Tuple, Union
+
+import librosa
+import numpy as np
+import torch
+
+from nemo.collections.asr.parts.preprocessing.segment import AudioSegment
+from nemo.collections.asr.parts.utils.audio_utils import ChannelSelectorType
+from nemo.collections.common.parts.preprocessing import collections
+from nemo.core.classes import Dataset
+from nemo.core.neural_types import AudioSignal, EncodedRepresentation, LengthsType, NeuralType
+from nemo.utils import logging
+from nemo.utils.decorators import experimental
+
+__all__ = [
+    'AudioToTargetDataset',
+    'AudioToTargetWithReferenceDataset',
+    'AudioToTargetWithEmbeddingDataset',
+]
+
+
+# Local utility functions
+def flatten_iterable(iter: Iterable[Union[str, Iterable[str]]]) -> Iterable[str]:
+    """Flatten an iterable which contains strings or
+    iterable with strings.
+
+    Args:
+        iter: iterable containing strings at deepest level
+
+    Returns:
+        Flat iterable with strings.
+    """
+    for it in iter:
+        if isinstance(it, str):
+            yield it
+        else:
+            yield from flatten_iterable(it)
+
+
+def flatten(list_in: List[Union[str, List[str]]]) -> List[str]:
+    """Flatten a list of (nested lists of) strings into a list.
+
+    Args:
+        list_in: list of strings, possibly nested
+
+    Returns:
+        List of strings, flat.
+    """
+    return list(flatten_iterable(list_in))
+
+
+def load_samples(
+    audio_file: str,
+    sample_rate: int,
+    duration: Optional[float] = None,
+    channel_selector: ChannelSelectorType = None,
+    fixed_offset: float = 0,
+) -> np.ndarray:
+    """Load samples from an audio file.
+
+    Args:
+        audio_file: path to an audio file
+        sample_rate: desired sample rate for output samples
+        duration: Optional desired duration of output samples.
+                  If `None`, complete file will be loaded.
+                  If set, a random segment of `duration` seconds will be
```

Comment:
It would be great of the function docsting explains more about optional multichannel aspect.
I think diarization/buffer-ASR codes should use this function for removing duplications